roomserver/internal/input/input_events.go

```diff
@@ -451,7 +451,7 @@ func (r *Inputer) processRoomEvent(
 	}
 
 	// Handle remote room upgrades, e.g. remove published room
-	if event.Type() == "m.room.tombstone" && event.StateKeyEquals("") && !r.Cfg.Matrix.IsLocalServerName(senderDomain) {
+	if event.Type() == "m.room.tombstone" && event.StateKeyEquals("") && !r.Cfg.Matrix.IsLocalServerName(sender.Domain()) {
```

Comment:
`sender` is obtained via `event.UserID()` which is the wrong way to check for "is this user local". We should be bouncing the `event.SenderID()` via `QueryUserIDForSender`.

Reply:
Come to think of it, it would be lovely if `event.UserID()` returned `*spec.UserID` so it makes it clear that hey sometimes this will be `nil`, which in and of itself is not an error.